hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheNearCacheStateHolder.java

```diff
@@ -61,7 +62,7 @@ void prepare(CachePartitionSegment segment, Collection<ServiceNamespace> namespa
         int partitionId = segment.getPartitionId();
         partitionUuid = metaData.getOrCreateUuid(partitionId);
 
-        cacheNameSequencePairs = new ArrayList(namespaces.size());
+        cacheNameSequencePairs = Collections.synchronizedList(new ArrayList(namespaces.size()));
```

Comment:
from what I understand, it seems the issue is only regarding visibility of writes. I was wondering whether we need synchronization on the arrayList itself. What if we create a temporary variable, perform writes and then assign to volatile field. Same for reads. wdyt?